config/clients/js/template/client.mustache

```diff
@@ -0,0 +1,482 @@
+    const authorizationModelId = this.getAuthorizationModelId(options);
+    if (!authorizationModelId) {
+      throw new Error("authorization_model_id_required");
+    }
+    return this.api.readAuthorizationModel(authorizationModelId, options);
+  }
+
+  /**
+   * ReadLatestAuthorizationModel - Reads the latest authorization model for the current store
+   * @param options
+   */
+  async readLatestAuthorizationModel(options: ClientRequestOpts = {}): PromiseResult<ReadAuthorizationModelResponse> {
+    const { headers = {} } = options;
+    setHeaderIfNotSet(headers, CLIENT_METHOD_HEADER, "ReadLatestAuthorizationModel");
+    const authorizationModelsResponse = await this.readAuthorizationModels({ ...options, headers });
+    const response = authorizationModelsResponse as any as CallResult<ReadAuthorizationModelResponse>;
+    response.authorization_model = authorizationModelsResponse.authorization_models?.[0];
+    delete (response as any).authorization_models;
+    return response;
+  }
+
+  /***********************
+   * Relationship Tuples *
+   ***********************/
+
+  /**
+   * Read Changes - Reads the list of historical tuple writes and deletes
+   * @param clientReadChangesRequest
+   * @param {object} paginationOptions
+   * @param {number} [paginationOptions.pageSize]
+   * @param {string} [paginationOptions.continuationToken]
+   * @param options
+   */
+  async readChanges(clientReadChangesRequest: ClientReadChangesRequest, options: ClientRequestOptsWithAuthZModelId & PaginationOptions = {}): PromiseResult<ReadChangesResponse> {
+    return this.api.readChanges(clientReadChangesRequest.type, options.pageSize, options.continuationToken, options);
+  }
+
+  /**
+   * Read - Reads tuples previously written to the store (does not evaluate)
+   * @param clientReadRequest
+   * @param options
+   */
+  async read(clientReadRequest: ClientReadRequest, options: ClientRequestOpts = {}): PromiseResult<ReadResponse> {
+    return this.api.read({ tuple_key: clientReadRequest }, options);
+  }
+
+  /**
+   * Write - Create or delete relationship tuples
+   * @param {object} body
+   * @param {objects} options
+   */
+  async write(body: ClientWriteRequest, options: ClientRequestOptsWithAuthZModelId & ClientWriteRequestOpts = {}): Promise<ClientWriteResponse> {
+    const { transaction = {}, headers = {} } = options;
+    const { maxPerChunk = 1 } = transaction; // 1 has to be the default otherwise the chunks will be sent in transactions
+    const { writes, deletes } = body;
+    const authorizationModelId = this.getAuthorizationModelId(options);
+
+    if (!transaction?.disable) {
+      await this.api.write({
+        writes: { tuple_keys: writes || [] },
+        deletes: { tuple_keys: deletes || [] },
+        authorization_model_id: authorizationModelId,
+      }, options);
+      return {
+        writes: writes?.map(tuple => ({
```

Comment:
We do that for the success because we'd like the response interface to be the same in both cases, we don't do that for the failure because it is an error in the transaction mode.

Reply:
If that's the case, the caller will need to update how they check for error in the case of switching between transaction vs. non-transaction

Reply:
Updated the README for that